src/main/java/uk/gov/digital/ho/hocs/casework/migration/api/MigrationCaseDataService.java

```diff
@@ -0,0 +1,83 @@
+package uk.gov.digital.ho.hocs.casework.migration.api;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.digital.ho.hocs.casework.api.dto.CaseDataType;
+import uk.gov.digital.ho.hocs.casework.client.auditclient.AuditClient;
+import uk.gov.digital.ho.hocs.casework.client.infoclient.InfoClient;
+import uk.gov.digital.ho.hocs.casework.domain.exception.ApplicationExceptions;
+import uk.gov.digital.ho.hocs.casework.domain.model.CaseData;
+import uk.gov.digital.ho.hocs.casework.domain.repository.CaseDataRepository;
+
+import java.time.LocalDate;
+import java.util.Map;
+import java.util.UUID;
+
+import static net.logstash.logback.argument.StructuredArguments.value;
+import static uk.gov.digital.ho.hocs.casework.application.LogEvent.EVENT;
+import static uk.gov.digital.ho.hocs.casework.application.LogEvent.MIGRATION_CASE_NOT_FOUND;
+import static uk.gov.digital.ho.hocs.casework.application.LogEvent.MIGRATION_CASE_NOT_UPDATED_NULL_DATA;
+import static uk.gov.digital.ho.hocs.casework.application.LogEvent.MIGRATION_CASE_RETRIEVED;
+import static uk.gov.digital.ho.hocs.casework.application.LogEvent.MIGRATION_CASE_UPDATED;
+
+@Service
+@Slf4j
+public class MigrationCaseDataService {
+    protected final CaseDataRepository caseDataRepository;
+    protected final AuditClient auditClient;
+    protected final InfoClient infoClient;
+
+    public MigrationCaseDataService(CaseDataRepository caseDataRepository,
+                                    InfoClient infoClient,
+                                    AuditClient auditClient) {
+        this.caseDataRepository = caseDataRepository;
+        this.infoClient = infoClient;
+        this.auditClient = auditClient;
+    }
+
+    protected CaseData getCaseData(UUID caseUUID) {
+        log.debug("Getting Case: {}", caseUUID);
+        CaseData caseData = caseDataRepository.findActiveByUuid(caseUUID);
+        if (caseData == null) {
+            log.error("Migration Case: {}, not found!", caseUUID, value(EVENT, MIGRATION_CASE_NOT_FOUND));
+            throw new ApplicationExceptions.EntityNotFoundException(String.format("Case: %s, not found!", caseUUID), MIGRATION_CASE_NOT_FOUND);
+        }
+        log.info("Got Migration Case: {}", caseData.getUuid(), value(EVENT, MIGRATION_CASE_RETRIEVED));
+        return caseData;
+    }
+
+    public void updateCaseData(UUID caseUUID, UUID stageUUID, Map<String, String> data) {
```

Comment:
This isn't used currently - I think this is fine with potential future use. If this isn't used we should remove this in the future.